### PR TITLE
Feature/add awscurl formula

### PR DESCRIPTION
**Pull request by okigan**
- [ x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Please squash all of your commits into one and fix the audit and style failures

**Comment by okigan** [comment]
@SMillerDev please advise on the remaining failing issues -- I think it's related to python dependencies install, but my attempts to resolve did not succeed

**Comment by okigan** [comment]
Bumping this up, opening for general comment: please advise on the remaining failing issues -- I think it's related to python dependencies install, but my attempts to resolve did not succeed

File: Formula/awscurl.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  desc "Curl like simplicity to access AWS resources"
+  homepage "https://github.com/okigan/awscurl"
+  url "https://github.com/okigan/awscurl/archive/ec83491.tar.gz"
+  version "ec83491d6b1e123af1bf100947731e113a32468d"
+  sha256 "0faef7dc822cef9c423185ed176636b710954bdaa36218f0d40d1a73ad444dab"
+  head "https://github.com/okigan/awscurl.git"
+
+  # TODO: If you're submitting an existing package, make sure you include your
+  # bottle block here.
```
Comment (jonchang) [inline Formula/awscurl.rb]: Remove these comments.

File: Formula/awscurl.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    sha256 "f3c5fd51747d450d4dcf6f923c81f78f811aab8205fda64b0aba34a4e48b0745"
+  end
+
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match "Curl", shell_output("#{bin}/awscurl --help")
```
Comment (jonchang) [inline Formula/awscurl.rb]: Test failures from CI:

```
==> /usr/local/Cellar/awscurl/ec83491d6b1e123af1bf100947731e113a32468d/bin/awscurl --help
Traceback (most recent call last):
  File "/usr/local/Cellar/awscurl/ec83491d6b1e123af1bf100947731e113a32468d/bin/awscurl", line 11, in <module>
    load_entry_point('awscurl==0.19', 'console_scripts', 'awscurl')()
TypeError: main() missing 1 required positional argument: 'argv'
Error: awscurl: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<1>.
```

**Review by jonchang — CHANGES_REQUESTED** [review]
The test failure looks like an upstream issue. Generally `main()` should not take arguments in a Python script, and that's likely what's causing the test failure.

Additionally, there were audit failures:

```
        Error: 2 problems in 1 formula detected
awscurl:
  * C: 14: col 1: Extra blank line detected.
  * C: 77: col 1: Extra blank line detected.
```

**Comment by chenrui333** [comment]
Relates to #48129

**Pull request closed without merging**